python/lsst/the/monster/isolate_and_transform.py

```diff
@@ -0,0 +1,149 @@
+import os
+from smatch import Matcher
+
+from lsst.pipe.tasks.isolatedStarAssociation import IsolatedStarAssociationTask
+
+from lsst.the.monster.splinecolorterms import ColortermSpline
+from lsst.the.monster.refcats import GaiaXPInfo, GaiaDR3Info, SkyMapperInfo, PS1Info, VSTInfo
+from lsst.the.monster.utils import read_stars
+
+__all__ = ["MatchAndTransform"]
+
+"""
+This Python script takes a sharded catalog and outputs a shard transformed to
+DES bandpasses.
+
+The following steps are performed:
+1. Read in the Gaia shard.
+2. Run the IsolatedStarAssociationTask on the Gaia shard.
+3. Read in each catalog.
+4. Match each catalog with the Gaia DR3 catalog.
+5. For each band, interpolate the DES magnitudes.
+6. Save the catalog shard.
+"""
+
+
+class MatchAndTransform:
+    """Match catalogs to Gaia and transform them to 'the_monster'
+       reference frame.
+
+    Parameters
+    ----------
+    htmid : `int`
+        htm level 7 id of catalog(s).
+    catalog_list : `list`
+        List of RefcatInfo objects for catalogs to transform.
+    write_path_inp : `str`
+        (optional) The path to write the outputs to.
+    gaia_cat_info_class : `RefcatInfo object`
+        (optional) The input Gaia DR3 RefcatInfo object.
+    """
+    GaiaDR3CatInfoClass = GaiaDR3Info
+    testing_mode = False
+
+    def run(self,
+            htmid=None,
+            catalog_list=[GaiaXPInfo, SkyMapperInfo, PS1Info, VSTInfo],
+            write_path_inp=None,
+            gaia_cat_info_class=None
+            ):
```

Comment:
The run docstring should go here.  And should just be the htmid.  Everything else is for the class.